Issue-19: Add ManagerAccess entity, repository, and related enums

yvanhenang opened the pull request:
- Added ManagerAccess entity with fields for id, name, weight, status, managedAccessTypes, and scopeOfAccess.
          - Introduced validation constraints for weight (0 to 1).
          - Implemented ManagerAccessRepository for CRUD operations.
          - Defined enums ScopeOfManagerAccess and TypeOfManagedAccess to handle various access types and scopes.

yvanhenang committed:
Issue-19: Add ManagerAccess entity, repository, and related enums

          - Added ManagerAccess entity with fields for id, name, weight, status, managedAccessTypes, and scopeOfAccess.
          - Introduced validation constraints for weight (0 to 1).
          - Implemented ManagerAccessRepository for CRUD operations.
          - Defined enums ScopeOfManagerAccess and TypeOfManagedAccess to handle various access types and scopes.

NkwaTambe commented:
Hello @yvanhenang  can you please fix the PMD check that is failing

NkwaTambe commented on ledgers-bank-account-access-management/ledgers-bank-account-access-repository/src/main/java/de/adorsys/ledgers/baam/db/domain/SeniorManagerAccess.java:
```diff
@@ -0,0 +1,19 @@
+package de.adorsys.ledgers.baam.db.domain;
+
+import jakarta.persistence.*;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.AllArgsConstructor;
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class SeniorManagerAccess {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String name;
+}
```
Hello @yvanhenang These class has been implemented by someone else..

if you need these class to work on your own class, you can sync with @Koufan-De-King  on these..

NkwaTambe commented on ledgers-bank-account-access-management/ledgers-bank-account-access-repository/src/main/java/de/adorsys/ledgers/baam/db/domain/ScopeOfManagerAccess.java:
```diff
@@ -0,0 +1,8 @@
+    VIEW_ACCOUNT_DETAILS,
+    EXECUTE_TRANSACTIONS;
+
+}
```
We have an **enum** called **AccessScope** which defines the scope for all entity classes... do you think these should be here???

NkwaTambe commented:
Hello @yvanhenang base on these ticket requirements...
(

Accountability and Logging:

    Every action taken by a user with ManagerAccess is logged to maintain a robust audit trail. This includes:
        The creation of new access roles for other users.
        Modifications to existing access permissions.
        Suspensions or revocations of access.
    These logs are crucial for maintaining transparency and accountability, particularly in complex organizational structures.

)

please i think you should add some sort of **TimeStamp** implimentation

NkwaTambe commented on ledgers-bank-account-access-management/ledgers-bank-account-access-repository/src/test/java/de/adorsys/ledgers/baam/db/repository/ManagerAccessRepositoryTest.java:
```diff
@@ -0,0 +1,12 @@
+
+class ManagerAccessRepositoryTest {
+
+}
```
PLease right a full test for these entity..

use you can use an **h2 database**  which acts as an in memory DB for the test..

these is to ensure that the entire logic is working perfectly fine and everything is actually stored in a database.

Also please ensure to do a **git pull** so as to see recent changes, these will help you see the new recent project structure for our module.

yvanhenang committed:
Merge branch 'develop' into Issue-19-EntityOfManagerAccess

Koufan-De-King commented on ledgers-bank-account-access-management/ledgers-bank-account-access-repository/src/main/java/de/adorsys/ledgers/baam/db/domain/ManagerAccess.java:
```diff
@@ -0,0 +1,40 @@
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class ManagerAccess {
```
```ManagerAccess```, like all Access entities, is a subclass of ```BankAccountAccess``` and so this here it should extend it.
This will save you from having to rewrite many fields already present in the said superclass, such as ```weight``` and ```id```.

Koufan-De-King reviewed (changes requested):
Please @yvanhenang consider the changes addressed and make sure to pull the most recent changes before refactoring, as @NkwaTambe sugested.

yvanhenang committed:
Issue-19: Refactor ManagerAccess domain and repository tests

- Updated ManagerAccess and ExampleManagerAccess classes for better test coverage.
- Enhanced test cases in ManagerAccessRepositoryIT.java and ManagerAccessRepositoryTest.java.
- Modified ScopeOfManagerAccess for improved functionality and validation.
